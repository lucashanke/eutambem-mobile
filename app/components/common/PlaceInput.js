import React from 'react';
import PropTypes from 'prop-types';
import { Keyboard } from 'react-native';

import { TextInput } from './TextInput';
import input from '../hoc/input';

const PlaceInput = (props) => {
  const onPlaceChange = (item) => {
    props.onValueChange(item);
    props.navigation.goBack(null);
  };

  return (
    <TextInput
      {...props}
      onFocus={() => {
        Keyboard.dismiss();
        props.navigation.navigate('PlaceScreen', {
          title: props.placeScreenTitle,
          types: props.types,
          onPlaceChange,
        });
      }}
    />
  );
};

PlaceInput.propTypes = {
  placeScreenTitle: PropTypes.string,
  types: PropTypes.arrayOf(PropTypes.oneOf(['establishment', 'address'])).isRequired,
  onValueChange: PropTypes.func.isRequired,
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
    goBack: PropTypes.func,
  }).isRequired,
};

PlaceInput.defaultProps = {
  placeScreenTitle: 'Informe o local',
};

export default input(PlaceInput);
