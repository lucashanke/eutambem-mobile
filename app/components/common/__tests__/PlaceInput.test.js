import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { PlaceInput } from '../PlaceInput';

const props = {
  placeScreenTitle: 'Inform the place',
  types: ['establishment'],
  onValueChange: sinon.spy(),
  navigation: {
    navigate: sinon.spy(),
    goBack: sinon.spy(),
  },
  placeholder: 'Place',
};

describe('PlaceInput', () => {
  it('renders a TextInput', () => {
    const wrapper = shallow(<PlaceInput {...props} />);
    expect(wrapper.find('TextInput')).toHaveLength(1);
  });

  it('focus of the TextInput calls the navigate method to the PlacePicker screen', () => {
    const wrapper = shallow(<PlaceInput {...props} />);

    wrapper.find('TextInput').simulate('focus');

    expect(props.navigation.navigate.calledWith('PlaceScreen')).toBeTruthy();
  });

  it('navigation passes onPlaceChange function that calls onValueChange prop', () => {
    const wrapper = shallow(<PlaceInput {...props} />);

    const { onPlaceChange } = wrapper.props().navigation.navigate.args[0][1];
    onPlaceChange({ id: 'testPlace', label: 'place for testing' });

    expect(props.onValueChange.callCount).toEqual(1);
    expect(props.onValueChange.calledWith({
      id: 'testPlace',
      label: 'place for testing',
    })).toBeTruthy();
    expect(props.navigation.goBack.callCount).toEqual(1);
  });
});
