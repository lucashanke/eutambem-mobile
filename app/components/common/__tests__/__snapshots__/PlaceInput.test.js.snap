// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceInput renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <ForwardRef(View)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <PlaceInput
      navigation={
        Object {
          "goBack": [Function],
          "navigate": [Function],
        }
      }
      onValueChange={[Function]}
      placeScreenTitle="Inform the place"
      placeholder="Place"
      required={false}
      types={
        Array [
          "establishment",
        ]
      }
    />
  </ForwardRef(View)>
</ForwardRef(View)>
`;
